### ISSUE TITLE
Build a Chimoney Slack Bot

### ISSUE THREAD

**Issue by uchibeke** [Chimoney/chimoney-community-projects#45]
HR, Event organizers, community managers and anyone in a Slack org should be able to send Chimoney to other users within Slack

Build a bot that rewards Slack users with Chimoney

- Bot can be mentioned in a channel or issue and instructed to send _$N_ to a _@user_ using Chimoney. 
- Bot [initiates a Chimoney payout](https://chimoney.readme.io/reference/post_v0-2-payouts-initiate-chimoney) to the email of the Slack user that is mentioned ([get Slack profile by username](https://api.slack.com/methods/users.profile.get)) and sends the initiate chimoney payout with `email: email_from_slack` 
- Bot sends a DM to the Slack user that initiated the Payout with a payment link to pay for the transaction on Chimoney. This link is returned from the `/initiate-chimoney` endpoint.
- Payer sees a Table on Slack build with Slack UI components with a summary of the Payout
- The Payer user clicks the link and is taken to the chimoney Payment page and they can pay from their Chimoney wallet
- Chimoney automatically sends an Email to the user that was rewarded. Optionally, reply to the original Slack comment or send a DM to the Slack User that received the Payment with the Redeem instructions that was also sent by Chimoney via email
- They can then redeem their reward to any of the options at [dash.chimoney.io/redeem](https://dash.chimoney.io/redeem)

**Comment by fumblehool** [comment]
Hello @uchibeke I would love to work on this.
Please assign this to me if nobody is currently working on it 😄

**Comment by thelimeskies** [comment]
Hi @uchibeke i would love to work on this if it's available

**Comment by uchibeke** [comment]
Done @thelimeskies and @fumblehool, feel free to implement it with your own approaches, languages/frameworks and both will qualify

**Comment by Ayo-Awe** [comment]
@uchibeke can I create a similar bot but for discord

**Comment by uchibeke** [comment]
> @uchibeke can I create a similar bot but for discord

Yes please. Please duplicate this issue for the discord bot and ping me on the new issue to assign it to you

let’s go! 🚀🚀🚀

**Comment by thelimeskies** [comment]
@uchibeke I'm almost done with the bot added a random giveaway feature for the channel it's called in 

it could currently be tested in my workspace would drop the link if that's ok with you

**Comment by uchibeke** [comment]
Can it be installed as an app in other workspaces? @thelimeskies

**Comment by thelimeskies** [comment]
Not at the moment 
Has an entire process I need to do for it to become distributed 

Will start the process once I'm done testing

**Comment by uchibeke** [comment]
Cool. Let me know

**Comment by mynameissaraah** [comment]
Hi  @uchibeke, I would like this to be assigned to me to work on.

**Comment by uchibeke** [comment]
@thelimeskies do you still want to complete this and earn?

**Comment by thelimeskies** [comment]
@uchibeke Sure, i've done some update and resolved the issues i had with state management the last time so i should push soon

**Comment by amosayomide05** [comment]
@uchibeke can I also work on it for a telegram bot?

**Comment by uchibeke** [comment]
> @uchibeke can I also work on it for a telegram bot?

Yes please

**Comment by amosayomide05** [comment]
> > @uchibeke can I also work on it for a telegram bot?
> 
> Yes please 

Ok thanks

**Comment by amosayomide05** [comment]
@uchibeke please i need help, i can't find my api key in sandbox

**Comment by amosayomide05** [comment]
@uchibeke I am done with the Telegram Bot

**Comment by amosayomide05** [comment]
@uchibeke @phyleria 
I am done creating the telegram bot

**Comment by phyleria** [comment]
Thanks @amosayomide05 

Your PR will be reviewed 👍🏾

**Comment by amosayomide05** [comment]
> Thanks @amosayomide05 
> 
> Your PR will be reviewed 👍🏾 

I haven't mad a PR yet

**Comment by phyleria** [comment]
If you're done creating the telegram bot then please create a PR for it

**Comment by phyleria** [comment]
Hey @amosayomide05 I'm creating the telegram bot issue and assigning you to it.

**Comment by amosayomide05** [comment]
> Oh, ok. If you're done creating the telegram bot then please create a PR for it

@phyleria I have created the PR #185

**Comment by amosayomide05** [comment]
> Oh, ok. If you're done creating the telegram bot then please create a PR for it

If I create a Whatsapp bot with similar feature, will it be part of the fest?

**Comment by uchibeke** [comment]
> > Oh, ok. If you're done creating the telegram bot then please create a PR for it
> 
> @phyleria If I create a Whatsapp bot with similar feature, will it be part of the fest?

yes please go ahead and create it and document it

What will be useful for all the bots you create is to do a short demo video, if possible

**Comment by amosayomide05** [comment]
> > > Oh, ok. If you're done creating the telegram bot then please create a PR for it
> > 
> > @phyleria If I create a Whatsapp bot with similar feature, will it be part of the fest?
> 
> yes please go ahead and create it and document it
> 
> What will be useful for all the bots you create is to do a short demo video, if possible

Ok
